grails-app/migrations/views/order-summary-view.sql

```diff
@@ -2,50 +2,46 @@ CREATE OR REPLACE VIEW order_item_status AS
     SELECT
         order_id,
         order_number,
-        product_code,
         quantity_ordered,
-        quantity_shipped,
+        SUM(quantity_shipped) AS quantity_shipped,
         order_item_id
     FROM (
-        SELECT order.id                 AS order_id,
+        SELECT
+            `order`.id                  AS order_id,
             `order`.order_number        AS order_number,
             order_item.id               AS order_item_id,
-            product.product_code        AS product_code,
-            order_item.quantity * order_item.quantity_per_uom    AS quantity_ordered, -- to compare with shipped quantity which is already multiplied by qty per uom
+            order_item.quantity         AS quantity_ordered,
             CASE
-                WHEN shipment.current_status IN ('SHIPPED', 'PARTIALLY_RECEIVED', 'RECEIVED') THEN SUM(shipment_item.quantity)
+                -- quantity divided by order_item.quantity_per_uom to match other quantities that are in uom
+                WHEN shipment.current_status IN ('SHIPPED', 'PARTIALLY_RECEIVED', 'RECEIVED') THEN SUM(shipment_item.quantity / order_item.quantity_per_uom)
                 ELSE 0
           	END AS quantity_shipped
         FROM `order`
             LEFT OUTER JOIN order_item ON order.id = order_item.order_id
-            LEFT OUTER JOIN product ON order_item.product_id = product.id
             LEFT OUTER JOIN order_shipment ON order_item.id = order_shipment.order_item_id
             LEFT OUTER JOIN shipment_item ON shipment_item.id = order_shipment.shipment_item_id
             LEFT OUTER JOIN shipment ON shipment.id = shipment_item.shipment_id
         WHERE `order`.order_type_id = 'PURCHASE_ORDER' AND order_item.order_item_status_code != 'CANCELLED'
         GROUP BY `order`.id, order_item.id, shipment.id
     )
-AS order_item_status;
+AS order_item_status GROUP BY order_item_id;
 
 CREATE OR REPLACE VIEW order_receipt_status AS
     SELECT
         order_id,
         order_item_id,
         order_number,
-        quantity_ordered,
-        quantity_received,
-        quantity_canceled
+        SUM(quantity_received) AS quantity_received,
+        SUM(quantity_canceled) AS quantity_canceled
```

Comment:
What's the reason behind removing quantity ordered. Are we just getting that from a different view?

Reply:
I think I wanted to base the quantity ordered on the order item summary, and not have the same quantities in each of these views and keep each "minor" view tied to specific values:
- **order_item_status** for quantity_ordered and quantity_shipped,
- **order_receipt_status** for quantity_received and quantity_canceled,
- **order_item_payment_status** for quantity_invoiced for order items,
- **order_adjustment_payment_status** for quantity_ordered (which is used for adjustment items count), quantity_invoiced (which is used for adjustment items invoiced count)